src/MensaWidget/MWFrame.class.st

```diff
@@ -67,48 +62,123 @@ MWFrame class >> scrollBarThickness [
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'tmc 6/6/2024 21:37'
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 20:55'
 }
-MWFrame >> buildFrameContent [
+MWFrame >> buildContentMorph [
 
-	| contentMorph |
-	
-	contentMorph := ScrollPane new.
-	self fillContentMorph: contentMorph.
-	contentMorph scrollBarThickness: self class scrollBarThickness;
-		color: self class contentMorphColor;
+	^ AlignmentMorph newColumn
+		cellInset: self class cellSpacing;
+		hResizing: #shrinkWrap;
+		vResizing: #shrinkWrap
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 16:21'
+}
+MWFrame >> buildLayout [
+
+	^ AlignmentMorph newColumn
+		hResizing: #spaceFill;
+		vResizing: #spaceFill;
 		position: self class contentPosition;
-		extent: self class contentExtent.
-	self addPaneMorph: contentMorph
+		extent: self class contentExtent;
+		color: MWColorTheme backgroundColor;
+		addMorph: self buildScrollPane;
+		addMorph: self buildMenuBar;
+		yourself
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 6/7/2024 11:05'
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 15:51'
 }
-MWFrame >> fillContentMorph: aContentMorph [
+MWFrame >> buildMenuBar [
 	
+	^ self dayPicker: (MWMenuBar new target: self);
+		dayPicker
+]
+
+{
+	#category : #creation,
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 16:21'
+}
+MWFrame >> buildScrollPane [
+
+	^ self contentMorph: 
+		(ScrollPane new 
+			scrollBarThickness: self class scrollBarThickness;
+			color: MWColorTheme backgroundColor;
+			extent: self class contentExtent;
+			yourself);
+		contentMorph
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 14:52'
+}
+MWFrame >> contentMorph [
+
+	^ contentMorph
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/7/2024 14:52'
+}
+MWFrame >> contentMorph: anObject [
+
+	contentMorph := anObject
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 15:54'
+}
+MWFrame >> dayPicker [
+
+	^ dayPicker
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 15:54'
+}
+MWFrame >> dayPicker: anObject [
+
+	dayPicker := anObject
+]
+
+{
+	#category : #dayManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/9/2024 15:53'
+}
+MWFrame >> displayDay [
+
 	| contentMorph |
 	
-	contentMorph := AlignmentMorph newColumn
-		cellInset: self class cellSpacing;
-		hResizing: #shrinkWrap;
-		vResizing: #shrinkWrap.
-	MWDataGettingWrapper getTodaysMeals do: [:each | 
-		contentMorph addMorph: each getCanteenMorph].
-	aContentMorph scroller addMorph: contentMorph
+	"delete old content"
+	self contentMorph scroller submorphsDo: [:each | each abandon].
+	"create new content"
```

Comment:
as the comments state, this method actually does two things, delete the old content and create the new content. Regarding separation of concerns i think it would be cleaner to have a method for each functionality and then this method that just uses both of them.